contentcuration/contentcuration/static/js/edit_channel/clipboard/views.js

```diff
@@ -1,196 +1,157 @@
-
-/* Loaded when user clicks edit icon on file*/
-var ClipboardEditFileView = Backbone.View.extend({
-	template: require("./hbtemplates/clipboard_edit_file.handlebars"),
-	initialize: function() {
-		_.bindAll(this, 'toggle_clipboard', 'update_file');
-		//this.listenTo(this.model, "change:number_of_hexagons", this.render);
+		this.listenTo(this.collection, "sync", this.render);
+        this.listenTo(this.collection, "remove", this.render);
 		this.render();
+		
 	},
 	render: function() {
-		this.$el.html(this.template({file: this.model.attributes}));
+		DragHelper.removeDragDrop(this);
+		this.$el.html(this.template({
+			collapsed : this.collapsed,
+			content_list : this.collection.toJSON()
+		}));
+		this.$el.find(".badge").html(this.collection.length);
+		
+		this.load_content();
+		this.$el.data("container", this);
+		this.$el.find("ul").data("list", this);
+		DragHelper.addDragDrop(this);
 	},
-	
 	events: {
-		'click .toggle_clipboard':'toggle_clipboard',
-		'click .clipboard_update_file':'update_file'
-	},
-	
-	toggle_clipboard: function(event){
-		$("#clipboard").hide();
+		'click .clipboard-toggler' : 'toggle_clipboard'
 	},
-	update_file: function(event){
-		if($("#content_name").val().trim() == "")
-				$("#name_err").css("display", "inline");
-		else{
-			//RELOAD TREE 
-			$("#clipboard").hide();
-		}
+	toggle_clipboard: function(){
+		this.collapsed = !this.collapsed;
+		if(this.collapsed) $("#clipboard").slideUp();
+		this.render();
+		if(!this.collapsed) $("#clipboard").slideDown();
+	},
+	load_content:function(){
+		var containing_list_view = this;
+		this.collection.forEach(function(entry){
+			var clipboard_item_view = new ClipboardItem({
+				containing_list_view: containing_list_view,
+				el: containing_list_view.$el.find("#clipboard_item_" + entry.id),
+				model: entry,
+				indent : 0,
+			});
+			containing_list_view.views.push(clipboard_item_view);
+		});
+	},
+	add_to_clipboard:function(models){
+		var container = this;
+		models.forEach(function(entry){
+			container.collection.add(entry);
+		});
+	},
+/*
+	drop_in_container: function(transfer){
+		console.log("adding");
+		var copy = transfer.data.model.duplicate(this.root.id);
+		this.collection.add(copy);
+		this.render();
+		console.log("added");
 	}
+	*/
 });
 
-/* Loaded when user clicks "Add Content" button */
-var ClipboardAddContentView = Backbone.View.extend({
-	template: require("./hbtemplates/clipboard_header.handlebars"),
-	initialize: function() {
-		_.bindAll(this, 'toggle_clipboard','computer_choose_content', 'channel_choose_content', 'choose_file', 'back_to_1', 'to_step_3','previous','preview_file','open_folder', 'close_file', 'open_folder_path', 'add_folder_to_list','add_tag','toggle_folder', 'clipboard_finish','choose_channel','add_file_to_list');
-		//this.listenTo(this.model, "change:number_of_hexagons", this.render);
+/* Loaded when user clicks clipboard button below navigation bar */
+var ClipboardItem = BaseViews.BaseListItemView.extend({
```

Comment:
To summarize my questions above - how much of this functionality is unique to ClipboardItem, and how much could be in the BaseListItemView?